### PR TITLE
Operations GRPC Implementation for Scavenging

### LINKED ISSUE #2151: Add Operations Proto Contract
The following operational operations need to be in a proto:

- Scavenge
- Subsystem Restarts

Furthermore, they should go into a separate project / nuget package e.g. `EventStore.Grpc.Operations.Client`

### PR DESCRIPTION
**Pull request by thefringeninja**
Part of #2151

### CODE REVIEW

File: src/EventStore.Core/Services/Storage/StorageScavenger.cs [hunk 1]
```diff
@@ -89,7 +89,7 @@ public class StorageScavenger :
 					} else {
 						message.Envelope.ReplyWith(new ClientMessage.ScavengeDatabaseResponse(message.CorrelationId,
 							ClientMessage.ScavengeDatabaseResponse.ScavengeResult.InvalidScavengeId,
-							_currentScavenge?.ScavengeId));
```
Comment (hayley-jean) [inline src/EventStore.Core/Services/Storage/StorageScavenger.cs]: Is there any particular reason why `currentScavenge.ScavengId` was changed to the `message.ScavengeId`?

Comment (thefringeninja) [reply src/EventStore.Core/Services/Storage/StorageScavenger.cs]: Yes, I would like to report back which scavengeid was invalid.

Comment (hayley-jean) [reply src/EventStore.Core/Services/Storage/StorageScavenger.cs]: Judging by the code around this, it looks like the original purpose was to return the currently running scavenge Id so if you sent the wrong one, you'd be able to get the correct one and be able to stop it in a subsequent call

Comment (thefringeninja) [reply src/EventStore.Core/Services/Storage/StorageScavenger.cs]: ok, makes sense. fixed now

**Comment by ahjohannessen** [comment]
Should there not be something like:

```
message GetStatusReq {
 	Options options = 1;
 	message Options {
 		string scavenge_id = 1;
 	}
 }

rpc GetStatus (GetStatusReq) returns (ScavengeResp);
```

File: src/EventStore.Core/ClusterVNodeStartup.cs [hunk 2]
```diff
@@ -94,7 +95,10 @@ public class ClusterVNodeStartup : IStartup, IHandle<SystemMessage.SystemReady>,
 								endpoint.MapGrpcService<Users>()))
 						.UseWhen(context => context.Request.Path.StartsWithSegments(StreamsSegment),
 							inner => inner.UseRouting().UseEndpoints(endpoint =>
-								endpoint.MapGrpcService<Streams>())),
+								endpoint.MapGrpcService<Streams>()))
+				.UseWhen(context => context.Request.Path.StartsWithSegments(OperationsSegment),  // TODO JPB figure out how to delete this sadness
```
Comment (ahjohannessen) [inline src/EventStore.Core/ClusterVNodeStartup.cs]: You guys should seriously consider using a different port for grpc stuff.

**Pull request merged**